lib/setup.c

```diff
@@ -921,6 +933,52 @@ ___SIZE_TS requested_words_still;)
 }
 
 
+___EXP_FUNC(___BOOL, ___fdset_resize)
+   ___P((int fd1,
+         int fd2),
+        (fd1,
+         fd2)
+int fd1;
+int fd2;)
+{
+#ifdef USE_select_or_poll
+
+  ___processor_state ___ps = ___PSTATE;
+  ___virtual_machine_state ___vms = ___VMSTATE_FROM_PSTATE(___ps);
+  ___sync_op_struct sop;
+
+  int newsize;
+  int fd = (fd2 > fd1) ? fd2 : fd1;
+
+  if (fd < ___vms->os.fdset.size)
+    return 1;
+
+  newsize = ___vms->os.fdset.size;
+  while (fd >= newsize)
+    newsize = ___CEILING_DIV (3 * newsize, 2);
+  /* align newsize to ___fdbits word boundaries */
+  newsize = (newsize + ___FDBITS - 1) & ~(___FDBITS - 1);
+
+  sop.op = OP_FDSET_CLEAROVL;
```

Comment:
Why is this still here?  There should only be a ```OP_FDSET_RESIZE```.  Hope I'm not reviewing stale changes!